pkg/awsutils/awsutils.go

```diff
@@ -698,6 +709,38 @@ func (cache *EC2InstanceMetadataCache) tagENI(eniID string) {
 	})
 }
 
+//parseAdditionalEniTagsMap will create a map for additional tags
+func parseAdditionalEniTagsMap(additionalEniTags string) (map[string]string, error) {
+	var additionalEniTagsMap map[string]string
+
+	// If duplicate keys exist, the value of the key will be the value of latter key.
+	err := json.Unmarshal([]byte(additionalEniTags), &additionalEniTagsMap)
+	if err != nil {
+		log.Errorf("Invalid format for ADDITIONAL_ENI_TAGS. Expected a json hash: %v", err)
+	}
+	return additionalEniTagsMap, err
+}
+
+// MapToTags converts a map to a slice of tags.
+func mapToTags(tagsMap map[string]string, tags []*ec2.Tag) []*ec2.Tag {
+	if tagsMap == nil {
+		return tags
+	}
+
+	for key, value := range tagsMap {
+		keyPrefix := "k8s.amazonaws.com"
+		if strings.Contains(key, keyPrefix) {
+			log.Warnf("Additional tags has %s prefix. Ignoring %s tag as it is reserved", keyPrefix, key)
+			continue
+		}
```

Comment:
Nit: Can we make this a constant instead, `reservedTagKeyPrefix = "k8s.amazonaws.com"` and also mention in the README that they will be ignored?